include/ck_tile/ops/epilogue/cshuffle_epilogue.hpp

```diff
@@ -387,6 +370,9 @@ struct CShuffleEpilogue
     // TODO: Check if there would be nicer ways to overload rather than with EmptyScale or nullptr_t
     struct EmptyScale
     {
+        // Include DataType here as EmptyScale will be utilized as a template parameter or in
+        // `operator()`, and DataType ensures EmptyScale adheres to the expected contract.
```

Comment:
You can use such work around for the concept:
    template <typename T>
    using has_data_type = decltype(ck::declval<T&>().GetDataType());